src/codegen/generators/statement_generator.rs

```diff
@@ -325,117 +328,102 @@ impl<'a, 'b> StatementCodeGenerator<'a, 'b> {
         body: &[AstNode],
     ) -> Result<(), Diagnostic> {
         let (builder, current_function, context) = self.get_llvm_deps();
-        self.generate_assignment_statement(counter, start)?;
-        let condition_check = context.append_basic_block(current_function, "condition_check");
-        let for_body = context.append_basic_block(current_function, "for_body");
-        let increment_block = context.append_basic_block(current_function, "increment");
-        let continue_block = context.append_basic_block(current_function, "continue");
-
-        //Generate an initial jump to the for condition
-        builder.build_unconditional_branch(condition_check);
-
-        //Check loop condition
-        builder.position_at_end(condition_check);
         let exp_gen = self.create_expr_generator();
-        let counter_statement = exp_gen.generate_expression(counter)?;
 
-        //.                                                           /            and_2                \
-        //.                  /             and 1               \
-        //.                   (counter_end_le && counter_start_ge) || (counter_end_ge && counter_start_le)
-        let or_eval = self.generate_compare_expression(counter, end, start, &exp_gen)?;
+        let end_ty = self.annotations.get_type_or_void(end, self.index);
+        let counter_ty = self.annotations.get_type_or_void(counter, self.index);
+        let ty = get_bigger_type(self.index.get_type_or_panic("DINT"), counter_ty, self.index);
+        let ll_ty = self.llvm_index.find_associated_type(ty.get_name()).unwrap();
```

Comment:
nit: Better names for `ty` and `ll_ty`? What does `ll` stand for?

Reply:
The `ll` stands for [llvm](https://llvm.moe/ocaml/Llvm.html).